editor/static/js/question/edit.js

```diff
@@ -771,20 +771,29 @@ $(document).ready(function() {
 
             //  Finally, mark duplicate names
             names.sort(Numbas.util.sortBy('name'));
+            var groupArray = [];
             function handle_group(group) {
-                group.forEach(function(n) {
-                    n.v.duplicateNameError(group.length > 1 ? n.name : null);
+                group.forEach(function (n) {
+                    n.v.duplicateNameError(null);
                 });
+                groupArray.push(group);
             }
 
             var start = 0;
-            names.forEach(function(n,i) {
-                if(n.name!=names[start].name) {
-                    handle_group(names.slice(start,i));
+            names.forEach(function (n, i) {
+                if (n.name != names[start].name) {
+                    handle_group(names.slice(start, i));
                     start = i;
                 }
             });
             handle_group(names.slice(start));
+            // Triggers duplicateNameError observable for groups with duplicates
+            groupArray.forEach(function (group) {
+                var groupVariable = group[0];
+                if (group.length > 1 && groupVariable.name != "") {
+                    groupVariable.v.duplicateNameError(groupVariable.name);
```

Comment:
You're only setting `duplicateNameError` on the first Variable in the group. You should set it for each variable in the group.

Again, `groupVariable` unnecessarily includes the type of the variable in its name. Just `group` will do.